msgraph/core/middleware/middleware.py

```diff
@@ -3,64 +3,44 @@
 # Licensed under the MIT License.
 # ------------------------------------
 import json
-import ssl
 
-from requests.adapters import HTTPAdapter
-from urllib3 import PoolManager
+import httpx
+from kiota_http.middleware import MiddlewarePipeline
 
-from .request_context import RequestContext
+from .request_context import GraphRequestContext
 
 
-class MiddlewarePipeline(HTTPAdapter):
-    """MiddlewarePipeline, entry point of middleware
-    The pipeline is implemented as a linked-list, read more about
-    it here https://buffered.dev/middleware-python-requests/
+class GraphRequest(httpx.Request):
```

Comment:
still needed?

Reply:
Static type checks were failing because `httpx.Request` does not have a `request_context` property. Created this custom type for graph requests

Reply:
but isn't the whole middleware structure already defined in kiota http already?

Reply:
I don't believe we should re-define concepts in graph core that already exist in kiota core libs. This is going to be really confusing to customers. Extending and specializing the existing, yes, redefining, no.

Reply:
It's only the default middleware pipeline that we are re-defining. For the middleware themselves, we are extending what is already in Kiota http

Reply:
I think the diff here is leading to confusion and git wrongly identified this as a modification when we're in fact talking about two different files. Is this Graph Request effectively what the core client defines for people to be able to execute arbitrary requests?